adds wheelchairs

Suicidalpickles opened the pull request:
:cl: SuicidalPickles
add: Added wheelchairs to the crafting menu, requires 2 plasteel and 8 metal rods.
/:cl:

this pr was made by disabled gang
adds wheelchairs, they have a movement delay of 2.5 at the moment.
![capture](https://user-images.githubusercontent.com/32374784/51150489-5b82e780-181b-11e9-94a0-fb507c732d8e.PNG)

ported from hippie station

Literallynotpickles committed:
ports wheelchair

BebeYoshi commented:
Can it be used by people without legs?

BebeYoshi commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,126 @@
+	max_integrity = 100
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 0, "bomb" = 10, "bio" = 0, "rad" = 0, "fire" = 20, "acid" = 30)
+	legs_required = 0
+	arms_requires = 0
```
Wait you can drive it while you are a nugget?

OverlyGeneric replied:
That seems like a consistency issue but makes sense if your being pushed by someone.

BebeYoshi replied:
Yeah, you should always move when being pushed, but you shouldn't be able to move it yourself without arms

subject217 replied:
Read further down in the code.

BebeYoshi replied:
True, my mistake then

Suicidalpickles commented:
> 
> 
> Can it be used by people without legs?

yes

duckay commented:
Didn't we have these ages ago but it was axed for some reason?

BebeYoshi reviewed (commented):
> Didn't we have these ages ago but it was axed for some reason?

I think it was axed because it was on the same pr as "hrp medbay"

Commit:
Update wheelchair.dm

Swiftkor commented:
This could make a pretty good quirk. Start with paralyzed legs or whatever and a wheelchair.

Then again, having tons of assistants in wheelchairs in every round might be a tad strange

subject217 commented:
actually having messed with it a bit - the code to stop it from moving is non-functional, and the arms_requires var doesn't appear to affect this

as a side note you should rename it to arms_required

Commit:
Update wheelchair.dm

Commit:
Update wheelchair.dm

Literallynotpickles committed:
wheelchair tweaks

Literallynotpickles committed:
wheelchair fix 2

Commit:
Update wheelchair.dm

Suicidalpickles commented:
> seems a bit expensive when scooters exist

didnt realize how cheap scooters are, could've sworn they were plasteel too. Gonna halve the wheelchair plasteel amount.

Commit:
Update recipes.dm

Suicidalpickles commented:
> as a side note you should rename it to arms_required

fixed that bit

> they're not shining paragons of code quality but you should look at how the scooter handles arm requirements

im bad with code so i just added the anti-nugget bit as a starter for now, can't figure out how to specifically have it disallow armless folk.

nicbn commented:
Why does it even need plasteel
I suggest replacing the plasteel requirement with cloth

nicbn commented:
Also you could make this a subtype of scooter (or make scooter a subtype of this) or something like that to avoid duplication of code

Tlaltecuhtli commented:
does require arms and legs even work? because i think you can drive just fine without legs

coiax commented:
When this gets merged, I'll make a Paraplegic quirk.

ExcessiveUseOfCobblestone commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+
+/obj/vehicle/ridden/wheelchair/buckle_mob(mob/living/H, force = 0, check_loc = 1)
+	if(!istype(H))
+		return 0
```
T/F

ExcessiveUseOfCobblestone commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+		return 0
+	if(H.get_num_legs() < 2 && H.get_num_arms() <= 0)
+		to_chat(H, "<span class='warning'>Your limbless body can't ride the [src].</span>")
+		return 0
```
T/F

ExcessiveUseOfCobblestone commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+	if(H.get_num_legs() < 2 && H.get_num_arms() <= 0)
+		to_chat(H, "<span class='warning'>Your limbless body can't ride the [src].</span>")
+		return 0
+	. = ..()
```
just use ..() if you're putting it at the bottom

ExcessiveUseOfCobblestone replied:
I only know byond and this (post review) is consistent with other parts of the code.

There's no point of setting `.` for implicit `return .` when you can just `return ..()` lol

ExcessiveUseOfCobblestone commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+	if(istype(I, /obj/item/wrench))
+		return
+	else
+		. = ..()
```
^

ExcessiveUseOfCobblestone commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+		unbuckle_mob(H)
+	H = null
+	user = null
+	. = ..()
```
just use ..() if you're putting it at the bottom

81Denton commented:
inb4 red-and-black syndicate wheelchairs that let you run people over

k-k-karma commented:
did you just fucking rip the code from hippie and remove all the fucking comments

k-k-karma commented:
you did not i am just special

deathride58 commented:
https://www.diffchecker.com/o2GrQAp5 Hey, credit your code if you're porting shit

Jujumatic commented:
If you're going to copy paste code and claim it as your own at least be intelligent enough to change the descriptions a little https://github.com/HippieStation/HippieStation/pull/7502/files

THIS IS FUCKEN DESGUSTEN!!!!!

ExcessiveUseOfCobblestone commented:
Don't do that again please.

deathride58 commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+	var/list/drive_sounds = list('sound/effects/roll.ogg')
+	var/mob/living/carbon/human/H
+	var/mob/living/user
+	movedelay = 8
```
This var doesn't even get used since you ditched Hippie's driver_move proc in favor of trying to make your code look original and brand new

deathride58 commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+	user = null
+	. = ..()
+
+/obj/vehicle/ridden/wheelchair/buckle_mob(mob/living/H, force = 0, check_loc = 1)
```
Just replace this proc with Hippie's driver_move proc. This here is just spaghetti and doesn't take into account the fact that people can lose limbs while riding the wheelchair like Hippie does.

deathride58 commented on code/modules/vehicles/wheelchair.dm:
```diff
@@ -0,0 +1,118 @@
+	overlays += O
+
+
+/obj/vehicle/ridden/wheelchair/proc/stopmove()
```
This proc also goes completely unused since you ditched Hippie's driver_move proc

Suicidalpickles commented:
im monky, not coderman

anyway ill scrap stuff in a bit

deathride58 commented:
The core of this case isn't that your code is bad. The issue that you straight-up plagiarized Hippiestation's code without even trying to point out that you ported it. [The diff between wheelchair.dm in your PR and wheelchair.dm in Hippiestation matches up almost perfectly line-for-line, with the only major difference being that you replaced the driver_move proc with a lackluster buckle_mob proc and made crafting wheelchairs take plasteel instead of metal](https://www.diffchecker.com/o2GrQAp5), so where's the "Ported from Hippiestation" attribution?

Suicidalpickles commented:
added credit in pr summary, like i said im a brainlet at this stuff

Thunder12345 commented:
♿ DISABLED GANG ROLLING THROUGH ♿

Suicidalpickles commented:
f2j489y312847`1-83 disabled gang will have to rest for now
